core/state_processor_test.go

```diff
@@ -354,3 +359,51 @@ func GenerateBadBlock(parent *types.Block, engine consensus.Engine, txs types.Tr
 	// Assemble and return the final block for sealing
 	return types.NewBlock(header, txs, nil, receipts, trie.NewStackTrie(nil), nil, true)
 }
+
+func TestStateProcessor_Process(t *testing.T) {
+	database := rawdb.NewMemoryDatabase()
+	gasLimit := uint64(math.MaxUint64)
+
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	auth, err := bind.NewKeyedTransactorWithChainID(key, params.TestingChainID)
+	require.NoError(t, err)
+
+	balance := big.NewInt(0).SetUint64(math.MaxUint64)
+	alloc := make(GenesisAlloc)
+	alloc[auth.From] = GenesisAccount{
+		Balance: balance,
+	}
+
+	cpcfg := params.TestChainConfig
+	cpcfg.ChainID = params.TestingChainID
+	genesis := Genesis{Config: cpcfg, GasLimit: gasLimit, Alloc: alloc, Coinbase: common.HexToAddress("0x0100000000000000000000000000000000000000")}
+	b := genesis.MustCommit(database)
+	cacheConfig := &CacheConfig{}
+	blockchain, _ := NewBlockChain(database, cacheConfig, genesis.Config, dummy.NewFaker(), vm.Config{}, common.Hash{})
+
+	chain, _, err := GenerateChain(genesis.Config, b, dummy.NewFaker(), database, 3, 10, func(i int, gen *BlockGen) {})
```

Comment:
Why 3 and 10?

Reply:
I changed `3` to `1` since I only needed one block. As for 10, it's sort of a block timestamp that affects the gas limits. In our test case, it can be any. I'll change it to `0` since it doesn't affect the test outcome.